server/server.go

```diff
@@ -423,3 +437,32 @@ func (s *Server) StartExitMonitor() {
 	}
 	<-done
 }
+
+func (s *Server) addContainerToWatcherWaitList(containerID string) error {
+	s.watcherLock.Lock()
+	defer s.watcherLock.Unlock()
+
+	if _, exist := s.watcherWaitList[containerID]; exist {
+		return fmt.Errorf("container %v already in the watcher ignore list", containerID)
```

Comment:
I think you want to update “watcher ignore list” → “watcher wait list” here.

And is there a problem with duplicate adds?  I wouldn't mind this being idempotent.

Reply:
Yes I have forgotten to update this comment...
About the check, I like to have it because I don't think we should have twice the same container being registered, and we could detect the error pretty soon.

Reply:
> About the check, I like to have it because I don't think we should have twice the same container being registered, and we could detect the error pretty soon.

What about parallel stop calls?  [Stops are idempotent][1], so the server could receive a second stop call while it is in the process of handling an earlier stop call for the same container.  You don't want to die there with “but I'm already waiting for that container to stop”.  I haven't thought that through all the way, so it may not be an issue.

[1]: https://github.com/kubernetes/kubernetes/blob/a77450ec2dc8fb25d8d6725bad8b70464f1ca6d2/pkg/kubelet/apis/cri/runtime/v1alpha2/api.proto#L51

Reply:
You got a point here, I'll make the change.

Reply:
done